src/Core/NotificationCenter/Commands/CreateNotificationCommand.cs

```diff
@@ -0,0 +1,41 @@
+﻿#nullable enable
+using Bit.Core.Context;
+using Bit.Core.Exceptions;
+using Bit.Core.NotificationCenter.Authorization;
+using Bit.Core.NotificationCenter.Commands.Interfaces;
+using Bit.Core.NotificationCenter.Entities;
+using Bit.Core.NotificationCenter.Repositories;
+using Microsoft.AspNetCore.Authorization;
+
+namespace Bit.Core.NotificationCenter.Commands;
+
+public class CreateNotificationCommand : ICreateNotificationCommand
+{
+    private readonly ICurrentContext _currentContext;
+    private readonly IAuthorizationService _authorizationService;
+    private readonly INotificationRepository _notificationRepository;
+
+    public CreateNotificationCommand(ICurrentContext currentContext,
+        IAuthorizationService authorizationService,
+        INotificationRepository notificationRepository)
+    {
+        _currentContext = currentContext;
+        _authorizationService = authorizationService;
+        _notificationRepository = notificationRepository;
+    }
+
+    public async Task<Notification> CreateAsync(Notification notification)
+    {
+        notification.SetNewId();
```

Comment:
⚠️ 

The generic repos already call this method.

So the ID set in the database will be different from what is set here.

It doesn't look like it is used in the command logic?

https://github.com/bitwarden/server/blob/c94a084c86f9e2a653e3e1562906a2ca8db321c9/src/Infrastructure.Dapper/Repositories/Repository.cs#L48

https://github.com/bitwarden/server/blob/c94a084c86f9e2a653e3e1562906a2ca8db321c9/src/Infrastructure.EntityFramework/Repositories/Repository.cs#L40

Reply:
Had no idea, thank you!
Fixed in afa22d17